packages/msw-addon/src/mswDecorator.js

```diff
@@ -1,23 +1,35 @@
-import { makeDecorator } from '@storybook/addons';
-import { setupWorker } from 'msw';
+import { initializeWorkerInstance } from './initializeWorker';
 
 let worker;
 
 export function initializeWorker(options) {
-  if (typeof global.process === 'undefined') {
-    worker = setupWorker();
-    worker.start(options);
+  worker = initializeWorkerInstance()
+  worker.start(options)
+
+  return worker
+}
+
+export function getWorker() {
+  if (worker === undefined) {
+    throw new Error(`[MSW] Tried to get worker but it was not defined yet. Did you forget to initialize it?`)
   }
+
+  return worker;
 }
 
-export const mswDecorator = makeDecorator({
-  name: 'withMsw',
-  parameterName: 'msw',
-  wrapper: (storyFn, context, { parameters = [] }) => {
-    if (worker) {
-      worker.resetHandlers();
-      worker.use(...parameters);
+export const mswDecorator = (storyFn, { parameters: { msw = [] } }) => {
+  if (worker) {
+    worker.resetHandlers();
+
+    if (!Array.isArray(msw)) {
+      throw new Error(`[MSW] expected to receive an array of handlers but received "${typeof msw}" instead.
+        Please refer to the documentation: https://mswjs.io/docs/getting-started/mocks/rest-api`)
```

Comment:
Assuming the user wants to mock a REST API is too strong. Let's direct them to the `/mocks` page in the docs where they will choose which API they want to mock.